JavaScript/JavaScriptSDK/Telemetry/Common/Envelope.ts

```diff
@@ -29,7 +29,7 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
                     return (this.sampleRate == 100) ? FieldType.Hidden : FieldType.Required;
                 },
                 tags: FieldType.Required,
-                data: FieldType.Required,
```

Comment:
They are not ok in IE7 and IE8 :cry:. Minimizer will remove them, but if you want to debug an unminified script, older IE's will throw an exception.

Fun fact: `[1,].length` returns 2 in IE7 :)